src/utils/AppRouteProvider.jsx

```diff
@@ -26,27 +60,45 @@ const AppRouteProvider = (props) => {
     ),
   };
 
+  const updateExperimentInfoOnNavigate = (projectUuid, experimentId) => {
+    const lastViewed = moment().toISOString();
+
+    dispatch(switchExperiment(experimentId));
+    dispatch(updateExperiment(experimentId, { lastViewed }));
+    dispatch(updateProject(projectUuid, { lastAnalyzed: lastViewed }));
+  };
+
   const continueNavigation = (nextRoute, hardNavigate) => {
     // Hard navigate, causing the page to refresh and fetch data from server
     if (hardNavigate) window.location.href = nextRoute;
     router.push(nextRoute);
   };
 
-  const handleRouteChange = (previousRoute, nextRoute, hardNavigate = false) => {
-    if (previousRoute.match('/data-processing') && changedQCFilters.size > 0) {
+  const handleRouteChange = (previousRoute, module, params, hardNavigate = false) => {
+    const nextRoute = PATHS[module].replace('[experimentId]', params.experimentId);
+
+    if (previousRoute.match(PATH_STUBS.DATA_PROCESSING) && changedQCFilters.size > 0) {
       setRenderIntercept(availableIntercepts.DATA_PROCESSING(nextRoute, hardNavigate));
       return;
     }
+
+    if (previousRoute.match(PATH_STUBS.DATA_MANAGEMENT)) {
+      // Update active project and experiment id when navigating from Data Management
+      const { projectUuid, experimentId } = params;
+      updateExperimentInfoOnNavigate(projectUuid, experimentId);
+    }
+
     continueNavigation(nextRoute, hardNavigate);
   };
 
   const navigateTo = (
-    nextRoute,
+    module,
+    params = {},
     refreshPage,
-  ) => handleRouteChange(router.pathname, nextRoute, refreshPage);
+  ) => handleRouteChange(router.pathname, module, params, refreshPage);
```

Comment:
So that the previous route (`router.pathname`) can be passed in as the first parameter. If we use `handleRouteChange` directly (in the component), then the component would know about the. Alternatively, we can have a global `router` variable in `AppRouteProvider` which `handleRouteChange` can access. However, I don't think it is a good idea as it would strongly couple `handleRouteChange` with `router`.

Reply:
But we have `router` already. The pathname can just be removed as an argument in `handleRouteChange` and just get it inside the function right?

Reply:
yea, but that's what I'm trying to avoid, using the global `router` variable, which introduces the coupling. It's also nice to have the `pathname` scoped locally inside the variable.